Assets/BossRoom/Scripts/Server/Game/Entity/ServerProjectileLogic.cs

```diff
@@ -92,8 +93,10 @@ private void FixedUpdate()
 
             if (m_DestroyAtSec < Time.fixedTime)
             {
-                // Time to go away.
-                Destroy(gameObject);
+                // Time return to the pool whence it came.
+                NetworkObject networkObject = gameObject.GetComponent<NetworkObject>();
+                NetworkObjectPool.Singleton.ReturnNetworkObject(networkObject, m_ProjectileInfo.ProjectilePrefab);
```

Comment:
I don't think this return is needed here. Calling Despawn will trigger `PooledPrefabInstanceHandler.HandleNetworkPrefabDestroy` which will return the object to the pool.